internal/pkg/agent/vault/vault_test.go

```diff
@@ -164,3 +174,73 @@ func TestVault(t *testing.T) {
 		t.Fatal(diff)
 	}
 }
+
+type secret struct {
+	Value     []byte    `json:"v"` // binary value
+	CreatedOn time.Time `json:"t"` // date/time the secret was created on
+}
+
+func TestVaultConcurrent(t *testing.T) {
+	const (
+		parallel   = 15
+		iterations = 7
+
+		key = `secret`
+	)
+
+	vaultPath := getTestVaultPath(t)
+
+	ctx, cn := context.WithCancel(context.Background())
+	defer cn()
+
+	for i := 0; i < iterations; i++ {
+		g, _ := errgroup.WithContext(context.Background())
+		for j := 0; j < parallel; j++ {
+			g.Go(func() error {
+				return doCrud(t, ctx, vaultPath, key)
+			})
+		}
+		err := g.Wait()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func doCrud(t *testing.T, ctx context.Context, vaultPath, key string) error {
```

Comment:
[Suggestion]
I don't see `t` being used, so you could remove it.
```suggestion
func doCrud(ctx context.Context, vaultPath, key string) error {
```